### DISCUSSION TITLE
How to implement Subscriptions with ExpressCheckout ?

### DISCUSSION THREAD

**Discussion by AnatoleT** [joshnuss/svelte-stripe#117] [Q&A] [answered]
Hello world :) 

I tried to implement a subscription system with svelte-stripe. It's actually working great with credit cards but that's not the case with some digital payment methods (like Apple Pay, Google Pay & Link).

Maybe someone can help me with this ? 
Happy coding :)

**Comment by joshnuss** [comment]
Hi @AnatoleT 

Have you looked at the Express Checkout example in the docs?

https://www.sveltestripe.com/examples/express-checkout

**Comment by AnatoleT** [comment]
Hi @joshnuss ! Thanks a lot for your quick answer and for your work on this one :)
Yes, I already implemented these for single payments, but I'm searching to implement subscriptions :)

**Reply by joshnuss** [reply]
Have you tried setting the `clientSecret` on the `Elements` container? I believe it's required for subscriptions to work

See example: https://github.com/joshnuss/sveltekit-stripe-subscriptions/blob/121ee13b55d54a830393ffc4594cb77617fddaae/src/routes/checkout/payment/%2Bpage.svelte#L40

**Reply by AnatoleT** [reply, marked as answer]
Thanks again for your help. I found what was the issue !
I was using Elements component with a fixed prop "mode" on "payment", i just had to switch it to "subscription" and it worked great. :)

**Reply by joshnuss** [reply]
Awesome, glad you figured it out!
